R/logRatioPlot.R

```
@@ -1,253 +1,565 @@
-                         scales = "free_y") {
-
-    assertthat::assert_that(plotType %in% c("bar", "point"),
-                            msg = "plotType must be either 'bar' or 'point'.")
-
-    .addGeoms <- function(myPlot){
-        if (tolower(plotType) == "bar") {
-            myPlot <- myPlot + geom_bar(stat = "identity",
-                                        alpha = barAlpha,
-                                        color = barColor,
-                                        fill = barFill,
-                                        size = barSize,
-                                        width = barWidth)
-        } else if (tolower(plotType) == "point") {
-            myPlot <- myPlot + geom_point(alpha = pointAlpha,
-                                          color = pointColor,
-                                          fill = pointFill,
-                                          size = pointSize,
-                                          shape = pointShape)
+                         labelAngle = 45,
+                         axisFree = TRUE) {
+    assertthat::assert_that(!missing(dgeObj),
+                            !is.null(dgeObj),
+                            "DGEobj" %in% class(dgeObj),
+                            msg = "dgeObj must be specified and should be of class DGEobj")
+
+    topTables <- DGEobj::getType(dgeObj, "topTable")
+    assertthat::assert_that(length(topTables) > 0,
+                            msg = "No topTable dataframes found in dgeObj. Please specify a dgeObj that contains topTable dataframes.")
+
+    assertthat::assert_that(all(sapply(topTables, class) == "data.frame"),
+                            msg = "topTables list must only contain dataframes.")
+
+    minNameLen <- min(sapply(names(topTables), nchar))
+    assertthat::assert_that(minNameLen > 0,
+                            msg = "All dataframes in topTable list must be named (it must be a named list.)")
+
+    geneData <- DGEobj::getItem(dgeObj, "geneData")
```

Can you change to getType. and also check if there are more than one geneData objects. If there are , we need to throw an error.( I guess. I will check with Connie tonight. Volcano plot which logesh is working on has the same set of steps for geneSymCol)

---

done